DFC.Digital/DFC.Digital.Web.Sitefinity.ContactUsModule/Mvc/Controllers/YourDetailsController.cs

```diff
@@ -119,29 +118,33 @@ public ActionResult SubmitDetails(ContactUsWithDobPostcodeViewModel viewModel)
         {
             if (ModelState.IsValid)
             {
-                var data = sessionStorage.Get() ?? new ContactUs();
-                var result = asyncHelper.Synchronise(() => sendEmailService.SendEmailAsync(new ContactUsRequest
-                {
-                    FirstName = viewModel.Firstname,
-                    Email = viewModel.EmailAddress,
-                    TemplateName = TemplateName,
-                    LastName = viewModel.Lastname,
-                    Message = data.ContactAnAdviserFeedback?.Message,
-                    TermsAndConditions = viewModel.AcceptTermsAndConditions,
-                    Postcode = viewModel.Postcode,
-                    ContactOption = data.ContactUsOption?.ContactOptionType.ToString(),
-                    ContactAdviserQuestionType = data.ContactAnAdviserFeedback?.ContactAdviserQuestionType.ToString(),
-                    DateOfBirth = viewModel.DateOfBirth
-                }));
-
-                if (result)
-                {
-                    return Redirect(SuccessPageUrl);
-                }
-                else
+                if (SessionDataValid())
                 {
+                    var data = sessionStorage.Get();
+                    var result = asyncHelper.Synchronise(() => sendEmailService.SendEmailAsync(new ContactUsRequest
+                    {
+                        FirstName = viewModel.Firstname,
+                        Email = viewModel.EmailAddress,
+                        TemplateName = TemplateName,
+                        LastName = viewModel.Lastname,
+                        Message = data.ContactAnAdviserFeedback?.Message,
+                        TermsAndConditions = viewModel.AcceptTermsAndConditions,
+                        Postcode = viewModel.Postcode,
+                        ContactOption = data.ContactUsOption?.ContactOptionType.ToString(),
+                        ContactAdviserQuestionType = data.ContactAnAdviserFeedback?.ContactAdviserQuestionType.ToString(),
+                        DateOfBirth = viewModel.DateOfBirth
+                    }));
+
+                    sessionStorage.Save(null);
```

Comment:
I think it would be cleaner to have a remove method on the sessionStorage rather than storing null value at controller level. Also think about whether session is valid functionality on the storage itself it it makes sense?